kong/dns/client.lua

```diff
@@ -0,0 +1,783 @@
+local cjson = require("cjson.safe")
+local utils = require("kong.dns.utils")
+local mlcache = require("kong.resty.mlcache")
+local resolver = require("resty.dns.resolver")
+
+local now       = ngx.now
+local log       = ngx.log
+local ERR       = ngx.ERR
+local WARN      = ngx.WARN
+local DEBUG     = ngx.DEBUG
+local ALERT     = ngx.ALERT
+local timer_at  = ngx.timer.at
+local worker_id = ngx.worker.id
+
+local pairs           = pairs
+local ipairs          = ipairs
+local math_min        = math.min
+local string_lower    = string.lower
+local table_insert    = table.insert
+local table_isempty   = require("table.isempty")
+
+local parse_hosts   = utils.parse_hosts
+local ipv6_bracket  = utils.ipv6_bracket
+local search_names  = utils.search_names
+local get_next_round_robin_answer           = utils.get_next_round_robin_answer
+local get_next_weighted_round_robin_answer  = utils.get_next_weighted_round_robin_answer
+
+local req_dyn_hook_run_hooks = require("kong.dynamic_hook").run_hooks
+
+
+-- Constants and default values
+
+local DEFAULT_ERROR_TTL   = 1     -- unit: second
+local DEFAULT_STALE_TTL   = 4
+local DEFAULT_EMPTY_TTL   = 30
+-- long-lasting TTL of 10 years for hosts or static IP addresses in cache settings
+local LONG_LASTING_TTL    = 10 * 365 * 24 * 60 * 60
+
+local PERSISTENT_CACHE_TTL = { ttl = 0 }  -- used for mlcache:set
+
+local DEFAULT_ORDER = { "LAST", "SRV", "A", "AAAA", "CNAME" }
+
+local TYPE_SRV      = resolver.TYPE_SRV
+local TYPE_A        = resolver.TYPE_A
+local TYPE_AAAA     = resolver.TYPE_AAAA
+local TYPE_CNAME    = resolver.TYPE_CNAME
+local TYPE_LAST     = -1
+
+local NAME_TO_TYPE = {
+  SRV     = TYPE_SRV,
+  A       = TYPE_A,
+  AAAA    = TYPE_AAAA,
+  CNAME   = TYPE_CNAME,
+  LAST    = TYPE_LAST,
+}
+
+local TYPE_TO_NAME = {
+  [TYPE_SRV]      = "SRV",
+  [TYPE_A]        = "A",
+  [TYPE_AAAA]     = "AAAA",
+  [TYPE_CNAME]    = "CNAME",
+  [TYPE_LAST]     = "LAST",
+}
+
+local HIT_L3 = 3 -- L1 lru, L2 shm, L3 callback, L4 stale
+
+local HIT_LEVEL_TO_NAME = {
+  [1] = "hit_lru",
+  [2] = "hit_shm",
+  [3] = "hit_cb",
+  [4] = "hit_stale",
+}
+
+-- server replied error from the DNS protocol
+local NAME_ERROR_CODE             = 3 -- response code 3 as "Name Error" or "NXDOMAIN"
+-- client specific error
+local CACHE_ONLY_ERROR_CODE       = 100
+local CACHE_ONLY_ERROR_MESSAGE    = "cache only lookup failed"
+local CACHE_ONLY_ANSWERS = { errcode = CACHE_ONLY_ERROR_CODE, errstr = CACHE_ONLY_ERROR_MESSAGE }
+local EMPTY_RECORD_ERROR_CODE     = 101
+local EMPTY_RECORD_ERROR_MESSAGE  = "empty record received"
```

Comment:
Let's remove `=` aligning. We have tried stop doing it, and we don't want it on new code.

```suggestion
local now = ngx.now
local log = ngx.log
local ERR = ngx.ERR
local WARN = ngx.WARN
local DEBUG = ngx.DEBUG
local ALERT = ngx.ALERT
local timer_at = ngx.timer.at
local worker_id = ngx.worker.id

local pairs = pairs
local ipairs = ipairs
local math_min = math.min
local string_lower = string.lower
local table_insert = table.insert
local table_isempty = require("table.isempty")

local parse_hosts = utils.parse_hosts
local ipv6_bracket = utils.ipv6_bracket
local search_names = utils.search_names
local get_next_round_robin_answer = utils.get_next_round_robin_answer
local get_next_weighted_round_robin_answer = utils.get_next_weighted_round_robin_answer

local req_dyn_hook_run_hooks = require("kong.dynamic_hook").run_hooks


-- Constants and default values

local DEFAULT_ERROR_TTL = 1  -- unit: second
local DEFAULT_STALE_TTL = 4
local DEFAULT_EMPTY_TTL = 30
-- long-lasting TTL of 10 years for hosts or static IP addresses in cache settings
local LONG_LASTING_TTL = 10 * 365 * 24 * 60 * 60

local PERSISTENT_CACHE_TTL = { ttl = 0 }  -- used for mlcache:set

local DEFAULT_ORDER = { "LAST", "SRV", "A", "AAAA", "CNAME" }

local TYPE_SRV = resolver.TYPE_SRV
local TYPE_A = resolver.TYPE_A
local TYPE_AAAA = resolver.TYPE_AAAA
local TYPE_CNAME = resolver.TYPE_CNAME
local TYPE_LAST = -1

local NAME_TO_TYPE = {
  SRV = TYPE_SRV,
  A = TYPE_A,
  AAAA = TYPE_AAAA,
  CNAME = TYPE_CNAME,
  LAST = TYPE_LAST,
}

local TYPE_TO_NAME = {
  [TYPE_SRV] = "SRV",
  [TYPE_A] = "A",
  [TYPE_AAAA] = "AAAA",
  [TYPE_CNAME] = "CNAME",
  [TYPE_LAST] = "LAST",
}

local HIT_L3 = 3 -- L1 lru, L2 shm, L3 callback, L4 stale

local HIT_LEVEL_TO_NAME = {
  [1] = "hit_lru",
  [2] = "hit_shm",
  [3] = "hit_cb",
  [4] = "hit_stale",
}

-- server replied error from the DNS protocol
local NAME_ERROR_CODE = 3 -- response code 3 as "Name Error" or "NXDOMAIN"
-- client specific error
local CACHE_ONLY_ERROR_CODE = 100
local CACHE_ONLY_ERROR_MESSAGE = "cache only lookup failed"
local CACHE_ONLY_ANSWERS = { errcode = CACHE_ONLY_ERROR_CODE, errstr = CACHE_ONLY_ERROR_MESSAGE }
local EMPTY_RECORD_ERROR_CODE = 101
local EMPTY_RECORD_ERROR_MESSAGE = "empty record received"
```

Reply:
Do we have this rule in contribute.md now?

Reply:
@chronolaw we could talk about this rule here, but I have resolved these `=` aligning.

Reply:
As @chronolaw has started a discussion thread in internal slack, so we could resolve this now!